### PR TITLE
Jormun vehicle positions

### PR DESCRIPTION
**Pull request by azime**
Jira: [NAVP-1861](https://jira.kisio.org/browse/NAVP-1861)

TODO:
- [x] Select VJs Only active at current datetime: in this [PR](https://github.com/CanalTP/navitia/pull/3569)

Json example for /vehicle_positions API:
image

### CODE REVIEW

**Review by benoit-bst — APPROVED** [review]
Whoop ! So nice to have a new end point 🍰

File: source/jormungandr/jormungandr/external_services/vehicle_position.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import logging
+from jormungandr.external_services.external_service import AbstractExternalService
+from jormungandr import utils
+import gevent
+import gevent.pool
+from jormungandr import cache, app
+from navitiacommon import type_pb2
+
+
+class VehcilePosition(AbstractExternalService):
```
Comment (pbench) [inline source/jormungandr/jormungandr/external_services/vehicle_position.py]:
```suggestion
class VehiclePosition(AbstractExternalService):
```

Comment (azime) [reply source/jormungandr/jormungandr/external_services/vehicle_position.py]: :+1:

File: source/jormungandr/jormungandr/external_services/external_service.py [hunk 2]
```diff
@@ -92,3 +92,20 @@ def _check_response(cls, response):
             if response.text:
                 error_msg += ' ({})'.format(response.text)
             raise ExternalServiceError(error_msg)
+
+    @classmethod
+    def response_marshaller(cls, response):
+        try:
```
Comment (xlqian) [inline source/jormungandr/jormungandr/external_services/external_service.py:98]: you can have only one `try catch` block

File: source/jormungandr/jormungandr/external_services/vehicle_position.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        futures = []
+        # TODO define new parameter forseti_pool_size ?
+        pool = gevent.pool.Pool(instance.realtime_pool_size)
+
+        # Copy the current request context to be used in greenlet
+        reqctx = utils.copy_flask_request_context()
+
+        def worker(vehicle_journey_position, args):
+            # Use the copied request context in greenlet
+            with utils.copy_context_in_greenlet_stack(reqctx):
```
Comment (xlqian) [inline source/jormungandr/jormungandr/external_services/vehicle_position.py:101]: :+1: :+1:

File: source/jormungandr/jormungandr/interfaces/v1/VehiclePositions.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        args = self.parsers["get"].parse_args()
+        instance = i_manager.instances.get(self.region)
+
+        args["filter"] = self.get_filter(split_uri(uri), args)
+        response = i_manager.dispatch(args, "vehicle_positions", instance_name=self.region)
+        instance.external_service_provider_manager.manage_vehicle_positions(
+            instance, response.vehicle_positions, **args
+        )
+        return response
+        #
```
Comment (xlqian) [inline source/jormungandr/jormungandr/interfaces/v1/VehiclePositions.py:99]:
```suggestion

```

File: source/jormungandr/jormungandr/external_services/external_service_provider_manager.py [hunk 5]
```diff
@@ -170,6 +170,15 @@ def manage_free_floatings(self, navitia_service, arguments):
...
         return service.get_response(arguments) if service else {'free_floatings': []}
 
+    # Here comes the function to call forseti/vehicle_positions
+    def manage_vehicle_positions(self, instance, response, **kwargs):
+
+        service = self._get_external_service("vehicle_positions")
+        if not service:
+            return
+        # Return empty object instead of None if error occurs while calling external service.
+        return service.update_response(instance, response, **kwargs)
```
Comment (xlqian) [inline source/jormungandr/jormungandr/external_services/external_service_provider_manager.py:180]:
```suggestion
        return service.update_response(instance, response, **kwargs) if service else None
```

**Review by xlqian — APPROVED** [review]

**Pull request merged**